src/submodule.c

```diff
@@ -149,6 +149,41 @@ static int find_by_path(const git_config_entry *entry, void *payload)
 	return 0;
 }
 
+static int can_add_submodule(git_repository *repo, const char *path) {
+	int error;
+	git_index *index;
+	git_buf dir = GIT_BUF_INIT;
+
+	if ((error = git_buf_sets(&dir, path)) < 0)
+		return error;
+
+	if ((error = git_path_to_dir(&dir)) < 0)
+		return error;
+
+	/* get the index for the repo */
+
+	if ((error = git_repository_index__weakptr(&index, repo)) < 0)
+		return error;
+
+	/* see if the submodule name exists as a file on the index */
+
+	if ((error = git_index_find(NULL, index, path)) == 0) {
```

Comment:
I think there's something off here. You first normalize `path` to a directory path by using `git_path_to_dir` and now you try to check if there's a file at that location matching the path. I guess what you wanted to do instead is first using the "normal" `path` value and only normalize it to a directory name after you've used `git_index_find` here to check whether there's a file there.